php-frontend/src/main/java/org/sonar/php/metrics/MetricsVisitor.java

```diff
@@ -0,0 +1,168 @@
+    List<Kind> result = new ArrayList<>(Arrays.asList(FUNCTION_NODES));
+    result.add(Kind.CLASS_DECLARATION);
+    result.add(Kind.COMPILATION_UNIT);
+    return result;
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (tree.is(Kind.COMPILATION_UNIT)) {
+      classComplexity = 0;
+      functionComplexity = 0;
+      functionComplexityDistribution = new RangeDistributionBuilder(CoreMetrics.FUNCTION_COMPLEXITY_DISTRIBUTION, LIMITS_COMPLEXITY_FUNCTIONS);
+      fileComplexityDistribution = new RangeDistributionBuilder(CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION, FILES_DISTRIBUTION_BOTTOM_LIMITS);
+      inputFile = fs.inputFile(fs.predicates().is(context().file()));
+
+    } else if (tree.is(Kind.CLASS_DECLARATION)) {
+      classComplexity += NewComplexityVisitor.complexity(tree);
+
+    } else if (tree.is(FUNCTION_NODES)) {
+      int currentFunctionComplexity = NewComplexityVisitor.complexityWithoutNestedFunctions(tree);
+      functionComplexity += currentFunctionComplexity;
+      functionComplexityDistribution.add(currentFunctionComplexity);
+    }
+  }
+
+  @Override
+  public void leaveNode(Tree tree) {
+    if (tree.is(Kind.COMPILATION_UNIT)) {
+      saveComplexityMetrics(context());
+      saveCounterMetrics(context());
+      saveLineMetrics(context());
+    }
+  }
+
+  private void saveCounterMetrics(CheckContext context) {
+// FIXME UNCOMMENT TO ENABLE COUNTER VISITOR
+//    CounterVisitor counter = new CounterVisitor(context.tree());
+//    saveMetricOnFile(CoreMetrics.FUNCTIONS, counter.getFunctionNumber());
+//    saveMetricOnFile(CoreMetrics.STATEMENTS, counter.getStatementsNumber());
+//    saveMetricOnFile(CoreMetrics.ACCESSORS, counter.getAccessorsNumber());
+//    saveMetricOnFile(CoreMetrics.CLASSES, counter.getClassNumber());
+  }
+
+  private void saveComplexityMetrics(CheckContext context) {
+    int fileComplexity = NewComplexityVisitor.complexity(context.tree());
+
+    saveMetricOnFile(CoreMetrics.COMPLEXITY, fileComplexity);
+    saveMetricOnFile(CoreMetrics.COMPLEXITY_IN_CLASSES, classComplexity);
+    saveMetricOnFile(CoreMetrics.COMPLEXITY_IN_FUNCTIONS, functionComplexity);
+
+    sensorContext.saveMeasure(inputFile, functionComplexityDistribution.build(true).setPersistenceMode(PersistenceMode.MEMORY));
+
+    fileComplexityDistribution.add(fileComplexity);
+    sensorContext.saveMeasure(inputFile, fileComplexityDistribution.build().setPersistenceMode(PersistenceMode.MEMORY));
+  }
+
+  private void saveLineMetrics(CheckContext context) {
+// FIXME UNCOMMENT TO ENABLE LINE VISITOR
+// NOTE Should be migrated before comment metrics
+//    LineVisitor lineVisitor = new LineVisitor(context.tree());
+//    int linesNumber = lineVisitor.getLinesNumber();
+//    Set<Integer> linesOfCode = lineVisitor.getLinesOfCode();
+//
+//    saveMetricOnFile(CoreMetrics.NCLOC, lineVisitor.getLinesOfCodeNumber());
+//    saveMetricOnFile(CoreMetrics.LINES, linesNumber);
+
+// FIXME UNCOMMENT TO ENABLE COMMENT VISITOR
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1134)